homeassistant/components/synology_dsm/__init__.py

```diff
@@ -127,9 +130,15 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         entry_data: SynologyDSMData = hass.data[DOMAIN][entry.unique_id]
         await entry_data.api.async_unload()
         hass.data[DOMAIN].pop(entry.unique_id)
+    hass.async_create_task(_notify_backup_listeners(hass), eager_start=False)
     return unload_ok
 
 
+async def _notify_backup_listeners(hass: HomeAssistant) -> None:
+    for listener in hass.data.get(SYNOLOGY_DATA_BACKUP_AGENT_LISTENERS, []):
+        listener()
+
```

Comment:
Looks like this doesn't need to be a coroutine.  I think the dispatcher could be used instead actually

Reply:
`async_dispatcher_connect` and `async_dispatcher_send` instead of a custom SYNOLOGY_DATA_BACKUP_AGENT_LISTENERS

Reply:
I'm struggling to bring this together with the need to wait for a config entry to be loaded, before the backup agent listener gets fired.

Reply:
I've switched back to the approach which is mentioned in the [backup](https://developers.home-assistant.io/docs/core/platform/backup) dev docs, since this is working.

I see an issue with any other approach as long as we don't have a defined method to listen for config entry state changes (_eq. becomes loaded or unloaded_), since the `backup.async_get_backup_agents` relies on `hass.config_entries.async_loaded_entries(DOMAIN)`. The approach from the [backup](https://developers.home-assistant.io/docs/core/platform/backup) dev docs is based on a task with `eager_start=False`, so until the task gets processed, the config entry is already fully loaded or unloaded